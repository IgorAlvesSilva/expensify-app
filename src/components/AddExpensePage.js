import React from 'react';
import ExpenseFrom from './ExpenseForm';
import {connect} from 'react-redux'
import {addExpense} from '../actions/expenses'


export class AddExpensePage extends React.Component {
  onSubmit = (expense => {
    this.props.addExpense(expense);
    this.props.history.push("/")
  })

  render() {
    return (
      <div>
      This is from my add expense component
      <ExpenseFrom 
        onSubmit={this.onSubmit}
      />
    </div>    
    )}}


const mapDispatchToProps = (dispatch) => {
  return {
    addExpense: (expense) => dispatch(addExpense(expense))
  };
}
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
