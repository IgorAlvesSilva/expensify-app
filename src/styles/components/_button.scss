.button {
    border: none;
    color: white;
    background-color: $blue;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}