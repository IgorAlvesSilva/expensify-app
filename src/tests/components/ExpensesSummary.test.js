import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test ('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={200} />);
    expect(wrapper).toMatchSnapshot();
});

test ('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={3798237} />);
    expect(wrapper).toMatchSnapshot();
});