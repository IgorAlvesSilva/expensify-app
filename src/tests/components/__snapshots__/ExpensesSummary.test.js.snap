// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.00
     
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense 2`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $37,982.37
     
  </h1>
</div>
`;
